pkg/exasol-rest-api/properties_file_parser.go

```diff
@@ -0,0 +1,45 @@
+package exasol_rest_api
+
+import (
+	"errors"
+	"gopkg.in/yaml.v3"
+	"os"
+)
+
+func getPropertiesFromFile(filepath string, properties *ApplicationProperties) error {
+	propertiesFile, err := openFile(filepath)
+	if err != nil {
+		return err
+	}
+	err = decodePropertiesFile(propertiesFile, properties)
+	if err != nil {
+		return err
+	}
+	closeFile(propertiesFile)
+	return nil
+}
+
+func openFile(filepath string) (*os.File, error) {
+	file, err := os.Open(filepath)
+	if err != nil {
+		return nil, err
+	} else if file == nil {
+		return nil, errors.New("properties file doesn't exist")
+	} else {
+		return file, nil
+	}
+}
+
+func decodePropertiesFile(propertiesFile *os.File, properties *ApplicationProperties) error {
+	decoder := yaml.NewDecoder(propertiesFile)
+	return decoder.Decode(&properties)
+}
+
+func closeFile(configFile *os.File) {
+	defer func(configFile *os.File) {
+		err := configFile.Close()
+		if err != nil {
+			errorLogger.Printf("error closing a file: %s. %s", configFile.Name(), err)
+		}
+	}(configFile)
```

Comment:
Not sure if `defer` works like this. I think this will run the code at the end of the `closeFile` function, but we want to close the file at the end of the `getPropertiesFromFile` function. So we need to use `defer` in `getPropertiesFromFile` after opening the file.

Reply:
I think it works as expected, otherwise I would have error while decoding the file if it's closed

Reply:
That's not what I wanted to say. Closing the file like this will work, but will `defer` is not necessary here. You could just remove it:
```suggestion
	err := configFile.Close()
	if err != nil {
		errorLogger.Printf("error closing a file: %s. %s", configFile.Name(), err)
	}
```
`defer` will run the code at the end of the function, but this is not helpful here. What you should do is use `defer` at the place right after you open the file: `defer closeFile(file)`. This will make sure that the file is closed even if there are parsing errors etc.

Reply:
Yes I think @kaklakariada is right here.

Reply:
So just remove the defer in this function